libsolidity/ast/Types.cpp

```diff
@@ -314,7 +314,7 @@ char const* Type::categoryName(Type::Category _category)
 	case Category::UserDefinedValueType: return "user-defined value type";
 	case Category::Tuple: return "tuple";
 	case Category::Mapping: return "mapping";
-	case Category::TypeType: return "type of a type";
+	case Category::TypeType: return "type";
```

Comment:
This is incorrect. Why was this changed?

Reply:
Oh, is it? AFAIK `TypeType` is stuff like a struct type, contract type, maybe even built-in types, like `uint`, righ? I got that wrong in #14087 so I'm fixing it but this did not seem important enough on its own to warrant a separate PR.

Reply:
No, it's not - how did you get that idea? All of those have their respective category. ``TypeType``, is the type of a type, as in ``type(SomeType)`` - which is considered as the type of ``SoneType``, i.e. the type of the type or TypeType.

I already wasn't happy about that other PR, since it has rather off nomenclature, but this is definitely, plainly incorrect.

Reply:
https://github.com/ethereum/solidity/pull/14087 already conceptualized this weirdly and inconsistently with anywhere else in the compiler.

Reply:
not sure, but I just write it here, because I was thinking about that after reading the comments.. isn't the type of a type a meta-type? maybe meta-type is a better name for `TypeType`? or is there any difference between a TypeType and a meta-type?